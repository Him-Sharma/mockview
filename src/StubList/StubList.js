import React from "react";
import StubItemList from "./StubItemList/StubItemList";
import "./StubList.css";

const stubList = () => {
  return (
    <div className="StubList">
      <StubItemList />
    </div>
  );
};

export default stubList;
