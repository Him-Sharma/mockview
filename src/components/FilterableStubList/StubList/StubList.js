import React from "react";
import "./StubList.css";

const StubList = (props) => {
  const filterText = props.filterText;
  console.log(`filter text is: ${filterText}`);
  return (
    <section className="stub-list">
      <ul>
        {props.stubs
          .filter((stub) => stub.predicates[0].equals.path === filterText)
          .map((stub) => (
            <li key={stub.id}>
              <span>{stub.predicates[0].equals.method}</span>
              <span>{stub.predicates[0].equals.path}</span>
            </li>
          ))}
      </ul>
    </section>
  );
};

export default StubList;
