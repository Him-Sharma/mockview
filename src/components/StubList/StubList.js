import React from "react";
import "./StubList.css";

const StubList = (props) => {
  return (
    <section className="stub-list">
      <ul>
        {props.stubs.map((stub) => (
          <li key={stub.id} onClick={props.selected.bind(this, stub.id)}>
            <span>{stub.method}</span>
            <span>{stub.endpointName}x</span>
          </li>
        ))}
      </ul>
    </section>
  );
};

export default StubList;
