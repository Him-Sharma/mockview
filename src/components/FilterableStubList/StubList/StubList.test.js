import React from "react";
import { render, cleanup } from "@testing-library/react";
import StubList from "./StubList";

afterEach(cleanup);

it("renders correctly", () => {
  const stubs = [
    {
      id: "0",
      predicates: [
        {
          equals: {
            method: "GET",
            path: "/posts",
          },
        },
      ],
      _links: {
        self: {
          href: "http://test",
        },
      },
    },
  ];
  const { baseElement } = render(
    <StubList stubs={stubs} filterText="/posts" />
  );
  expect(baseElement).toMatchSnapshot();
});
