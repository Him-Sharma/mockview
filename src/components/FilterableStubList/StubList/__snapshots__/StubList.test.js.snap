// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <section
      class="stub-list"
    >
      <ul>
        <li>
          <span>
            GET
          </span>
          <span>
            /posts
          </span>
        </li>
      </ul>
    </section>
  </div>
</body>
`;
